code/FromReferencesTask.cs

```diff
@@ -87,6 +101,19 @@ public override Boolean Execute()
 			}
 		}
 
+		private string GetDestinationPath(string sourceProjectBasePath, string sourceDirectoryPath)
+		{
+			Contract.Requires(sourceProjectBasePath != null);
+			Contract.Requires(sourceDirectoryPath != null);
+			Contract.Assume(sourceDirectoryPath.Length > sourceProjectBasePath.Length);
+			Contract.Assume(!String.IsNullOrEmpty(LibraryDirectoryFullPath));
+			var rootRelativePath = sourceDirectoryPath.Remove(0, sourceProjectBasePath.Length);
+			if (rootRelativePath[0] == '\\')
+			{
+				rootRelativePath = rootRelativePath.Substring(1);
+			}
+			return Path.Combine(LibraryDirectoryFullPath, rootRelativePath);
```

Comment:
If you change this to:
```cs
var combinedPath = Path.Combine(...);
Contract.Assume(combinedPath != null);
return combinePath;
```
then you will be able to add a `Contract.Ensures(Contract.Result<String>() != null)` to this method.  This will allow you to remove the `Contract.Assume(!String.IsNullOrEmpty(destinationPath));` from the other method above.

The problem is that `Path.Combine` doesn't have the correct contract on it (that it will never return null).